sim/rogue/items_sets_pve.go

```diff
@@ -213,3 +215,175 @@ var ItemSetNightSlayerBattlearmor = core.NewItemSet(core.ItemSet{
+		// Your Rolling with the Punches also grants you 20% increased Armor from items per stack.
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneRollingWithThePunches) {
+				return
+			}
+			initarmor := rogue.BaseEquipStats()[stats.Armor]
+
+			rogue.RegisterAura(core.Aura{
+				Label: "T2 Please Replace",
+				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+				  oldOnStacksChange := rogue.RollingWithThePunchesProcAura.OnStacksChange
+				  rogue.RollingWithThePunchesProcAura.OnStacksChange = func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+					oldOnStacksChange(aura, sim, oldStacks, newStacks)
+					rogue.AddStatDynamic(sim, stats.Armor, float64(0.2*initarmor*float64(newStacks-oldStacks)))
+				}
+				},
+			  })
+		},
+		// The cooldown on your Main Gauche resets every time your target Dodges or Parries.
+		6: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneMainGauche) {
+				return
+			}
+
+			core.MakePermanent(rogue.RegisterAura(core.Aura{
+				Label:    "S03 - Item - T2 - Rogue - Tank 6P Bonus",
+				OnSpellHitDealt: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if result.DidDodge() || result.DidParry() {
+						rogue.MainGauche.CD.Reset()
+					}
+				},
+			}))
+		},
+	},
+})
+
+var ItemSetMadCapsOutfit = core.NewItemSet(core.ItemSet{
+	Name: "Madcap's Outfit",
+	Bonuses: map[int32]core.ApplyEffect{
+		// +20 Attack Power
+		2: func(agent core.Agent) {
+			c := agent.GetCharacter()
+			c.AddStats(stats.Stats{
+				stats.AttackPower:       20,
+			})
+		},
+		// Increases your chance to get a critical strike with Daggers by 5%.
+		3: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			switch rogue.GetProcMaskForTypes(proto.WeaponType_WeaponTypeDagger) {
+			case core.ProcMaskMelee:
+				rogue.AddStat(stats.MeleeCrit, core.CritRatingPerCritChance*float64(5))
+			case core.ProcMaskMeleeMH:
+				// the default character pane displays critical strike chance for main hand only
+				rogue.AddStat(stats.MeleeCrit, core.CritRatingPerCritChance*float64(5))
+				rogue.OnSpellRegistered(func(spell *core.Spell) {
+					if spell.ProcMask.Matches(core.ProcMaskMeleeOH) {
+						spell.BonusCritRating -= core.CritRatingPerCritChance * float64(5)
+					}
+				})
+			case core.ProcMaskMeleeOH:
+				rogue.OnSpellRegistered(func(spell *core.Spell) {
+					if spell.ProcMask.Matches(core.ProcMaskMeleeOH) {
+						spell.BonusCritRating += core.CritRatingPerCritChance * float64(5)
+					}
+				})
+			}
+		},
+		// Increases the critical strike chance of your Ambush ability by 30%.
+		5: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.OnSpellRegistered(func(spell *core.Spell) {
+				if spell.SpellCode == SpellCode_RogueAmbush  {
+					spell.BonusCritRating += 30 * core.SpellCritRatingPerCritChance
```

Comment:
🔨 this should multiply by `core.CritRatingPerCritChance` just in case 😄 They're all 1 in SoD so it's just a knit

Reply:
Updated!